cudaaligner/tests/Test_NeedlemanWunschImplementation.cpp

```diff
@@ -172,12 +172,12 @@ matrix<int> ukkonen_gpu_build_score_matrix(const std::string& target, const std:
     std::vector<int32_t> path_length_h(1);
 
     device_buffer<char> sequences_d(2 * max_alignment_length, allocator);
-    GW_CU_CHECK_ERR(cudaMemcpy(sequences_d.data(), query.c_str(), sizeof(char) * query_length, cudaMemcpyHostToDevice));
-    GW_CU_CHECK_ERR(cudaMemcpy(sequences_d.data() + max_alignment_length, target.c_str(), sizeof(char) * target_length, cudaMemcpyHostToDevice));
+    cudautils::device_copy_n(query.c_str(), query_length, sequences_d.data());
+    cudautils::device_copy_n(target.c_str(), target_length, sequences_d.data() + max_alignment_length);
 
     device_buffer<int32_t> sequence_lengths_d(2, allocator);
-    GW_CU_CHECK_ERR(cudaMemcpy(sequence_lengths_d.data(), &query_length, sizeof(int32_t) * 1, cudaMemcpyHostToDevice));
-    GW_CU_CHECK_ERR(cudaMemcpy(sequence_lengths_d.data() + 1, &target_length, sizeof(int32_t) * 1, cudaMemcpyHostToDevice));
+    cudautils::device_copy_n(&query_length, 1, sequence_lengths_d.data());
+    cudautils::device_copy_n(&target_length, 1, sequence_lengths_d.data() + 1);
```

Comment:
Here you could actually use
https://github.com/clara-parabricks/GenomeWorks/blob/7a4e47eb1311bf8d4f2322ba139a102b9b91b137/common/base/include/claraparabricks/genomeworks/utils/cudautils.hpp#L132